# Case Study 1

### Thojing Dea Pallaqi

In [1]:
#Import Library Pandas dan Numpy
import pandas as pd
import numpy as np

In [2]:
#Import Dataset
df = pd.read_excel('Data/credit.xlsx')
df

Unnamed: 0   Catatan_kredit   Utang    Jaminan Pendapatan  Risiko
0             1            buruk  rendah  tidak_ada       0-15  tinggi
1             2  tidak_diketahui  tinggi  tidak_ada      15-35  tinggi
2             3  tidak_diketahui  rendah  tidak_ada      15-35  rendah
3             4            buruk  rendah  tidak_ada       0-15  tinggi
4             5  tidak_diketahui  rendah        ada        >35  rendah
..          ...              ...     ...        ...        ...     ...
115         116             baik  tinggi        ada      15-35  rendah
116         117             baik  tinggi  tidak_ada      15-35  rendah
117         118             baik  tinggi        ada      15-35  rendah
118         119  tidak_diketahui  rendah  tidak_ada      15-35  tinggi
119         120             baik  rendah  tidak_ada      15-35  tinggi

[120 rows x 6 columns]

Data yang digunakan pada studi kasus ini adalah credit.xlsx. 
Dataset memiliki 4 atribut (Catatan_kredit, Utang, Jaminan, dan Pendapatan) dengan nilai kategori yang 
berbeda-beda.

In [3]:
#Mengecek Kesesuaian Tipe Data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Unnamed: 0      120 non-null    int64 
 1   Catatan_kredit  120 non-null    object
 2   Utang           120 non-null    object
 3   Jaminan         120 non-null    object
 4   Pendapatan      120 non-null    object
 5   Risiko          120 non-null    object
dtypes: int64(1), object(5)
memory usage: 5.8+ KB


Berdasarkan hasil diatas data yang digunakan memiliki 6 Columns atau Variabel dan terdiri dari 120 data.
Untuk tipe data pada data tersebut adalah int64 dan object.

In [4]:
#Mengecek Missing Value
df.isna().sum()

Unnamed: 0        0
Catatan_kredit    0
Utang             0
Jaminan           0
Pendapatan        0
Risiko            0
dtype: int64

Setelah dilakukan pengecekan Missing Value atau data Hilang, dapat disimpulkan bahwa tidak ada Missing Value dari data tersebut.

In [5]:
#Import Library LabelEncoder
from sklearn.preprocessing import LabelEncoder

#Mengkonversi Variabel Menjadi Numerik
LEnconder = LabelEncoder()
columns = ['Catatan_kredit', 'Utang', 'Jaminan', 'Pendapatan']

for col in columns:
    df[col] = LEnconder.fit_transform(df[col])

In [6]:
df

Unnamed: 0  Catatan_kredit  Utang  Jaminan  Pendapatan  Risiko
0             1               1      0        1           0  tinggi
1             2               2      1        1           1  tinggi
2             3               2      0        1           1  rendah
3             4               1      0        1           0  tinggi
4             5               2      0        0           2  rendah
..          ...             ...    ...      ...         ...     ...
115         116               0      1        0           1  rendah
116         117               0      1        1           1  rendah
117         118               0      1        0           1  rendah
118         119               2      0        1           1  tinggi
119         120               0      0        1           1  tinggi

[120 rows x 6 columns]

Kemudian, dilakukan konversi variabel menjadi numerik dengan menggunakan LabelEnconder. Variabel yang dikonversi adalah variabel Catatan_kredit, Utang, Jaminan, dan Pendapatan.

In [7]:
#Membuang data yang tidak diperlukan
df.drop(columns= 'Unnamed: 0', inplace=True)
df.head()

Catatan_kredit  Utang  Jaminan  Pendapatan  Risiko
0               1      0        1           0  tinggi
1               2      1        1           1  tinggi
2               2      0        1           1  rendah
3               1      0        1           0  tinggi
4               2      0        0           2  rendah

Dikarenakan variabel Unnamed: 0 tidak digunakan dalam membangun model, maka variabel tersebut dihilangkan/dibuang dengan menggunakan drop.

In [8]:
#Import Library Split Data
from sklearn.model_selection import train_test_split

#Menentukan X dan Y
X = df.drop('Risiko', axis=1)
y = df['Risiko']

Kemudian, mengimport librari yang dibutuhkan untuk membagi data menjadi data train dan data test. Setelah itu, menentukan variabel X dan Y. Variabel X terdiri dari variabel Catatan_kredit, Utang, Jaminan, dan Pendapatan, sedangkan variabel Y adalah variabel risiko.

In [9]:
#Splitting data
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [10]:
print('x_train = ', X_train.shape)
print('x_test = ', X_test.shape)
print('y_train = ', y_train.shape)
print('y_test = ', y_test.shape)

x_train =  (96, 4)
x_test =  (24, 4)
y_train =  (96,)
y_test =  (24,)


Berikutnya, dilakukan pembagian atau split data menjadi data train dan data test dengan banyaknya data test 20% dari data yang ada. Hasil yang didapatkan adalah data train sebanyak 96 dan data test sebanyak 24.

In [11]:
# Import Library Modeling

from sklearn.neighbors import KNeighborsClassifier

In [12]:
# KNN
knn = KNeighborsClassifier(n_neighbors=2)
knn.fit(X_train, y_train)

KNeighborsClassifier(n_neighbors=2)

In [13]:
#Menerapkan model untuk prediksi dengan KNN
y_pred = knn.predict(X_test)
print("Predicted Values:")
print(y_pred)
print("Actual Values:")
print(y_test.values)

Predicted Values:
['tinggi' 'rendah' 'rendah' 'rendah' 'rendah' 'rendah' 'tinggi' 'tinggi'
 'tinggi' 'rendah' 'rendah' 'rendah' 'rendah' 'rendah' 'rendah' 'rendah'
 'rendah' 'tinggi' 'rendah' 'rendah' 'tinggi' 'tinggi' 'rendah' 'rendah']
Actual Values:
['tinggi' 'rendah' 'rendah' 'tinggi' 'rendah' 'rendah' 'rendah' 'tinggi'
 'tinggi' 'rendah' 'tinggi' 'rendah' 'tinggi' 'rendah' 'rendah' 'rendah'
 'tinggi' 'tinggi' 'rendah' 'tinggi' 'tinggi' 'tinggi' 'tinggi' 'rendah']


In [14]:
#Evaluasi Model yang terbentuk
from sklearn.metrics import confusion_matrix, classification_report
conf_matrix = confusion_matrix(y_test, y_pred)

print("Confusion Matrix:")
print(conf_matrix)


Confusion Matrix:
[[11  1]
 [ 6  6]]


In [15]:
class_report = classification_report(y_test, y_pred)

print("\nClassification Report:")
print(class_report)


Classification Report:
              precision    recall  f1-score   support

      rendah       0.65      0.92      0.76        12
      tinggi       0.86      0.50      0.63        12

    accuracy                           0.71        24
   macro avg       0.75      0.71      0.70        24
weighted avg       0.75      0.71      0.70        24



In [16]:
#Akurasi
knn.score(X_test, y_test)

0.7083333333333334